### ISSUE TITLE
[BUG][code-analyzer] False-positive on record-triggered flows: PreventPassingUserDataIntoElementWithoutSharing

### ISSUE THREAD

**Issue by jasonsiders** [forcedotcom/code-analyzer#1937]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report.

### Bug Description

Whenever a record-triggered/after-save flow contains DML or SOQL, the `PreventPassingUserDataIntoElementWithoutSharing` rule flags it:

The rule's [docs](https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/rules-flow.html) even call this out, and recommend setting the flow to 'User Mode' to prevent issues. However, the _How to Run the Flow_ dropdown is not available for record-triggered flows. 

### Output / Logs

```shell
jsiders@jsiders-mac salesforce % sf code-analyzer run --rule-selector flow --target force-app/main/default/flows/Opportunity_Closed_Lost_Update_Contact_to_Sales_Recycled.flow-meta.xml 

Streaming logs in real time to:
    /var/folders/8d/0t7sckgj10xcb32zycf088480000gp/T/sfca-2025_11_23_22_31_50_627.log

Selecting rules... done.
Executing rules... done. Executed rules from flow.

Violation file paths relative to '/Users/jsiders/Documents/VSCode/Nasuni/salesforce/force-app/main/default/flows'.
  #   Severity   Rule                                                   Location                                                                      Message                                                                                              
 ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  1   2 (High)   flow:PreventPassingUserDataIntoElementWithoutSharing   Opportunity_Closed_Lost_Update_Contact_to_Sales_Recycled.flow-meta.xml:75:1   User controlled data flows into recordUpdates element selector in run mode: SystemModeWithoutSharing



=== Summary

Found 1 violation(s) across 1 file(s):
    1 High severity violation(s) found.

Additional log information written to:
    /var/folders/8d/0t7sckgj10xcb32zycf088480000gp/T/sfca-2025_11_23_22_31_50_627.log
```

### Steps To Reproduce

1. Create a record-triggered flow, in an after-save (Actions & Related Records) context, that performs a query and/or DML operation
2. Retrieve the source of the new flow
3. Run `code-analyzer` against this new flow:
```sh
sf code-analyzer run --rule-selector flow --target {{path_to_the_flow-meta.xml_file}}
```

### Expected Behavior

This rule should not raise a violation if the flow's `<triggerType>` is `RecordAfterSave`.

### Operating System

macOS Tahoe 26.0

### Salesforce CLI Version

@salesforce/cli/2.112.6 darwin-arm64 node-v22.20.0

### Code Analyzer Plugin (code-analyzer) Version

code-analyzer 5.6.1

### Node Version

v22.20.0

### Java Version

openjdk version "17.0.13" 2024-10-15

### Python Version

Python 3.13.1

### Workaround

For now, we're getting around this by overriding the severity of the rule in `code-analyzer.yml`, but by doing so we lose the ability for our CI jobs to pick up _actual violations_ of this issue, rendering it more or less useless. 

### Urgency

Moderate

**Comment by rsussland** [comment]
Hi Jason, can you try running this with an updated version? I ran the file you provided and got this only, which is a valid issue:

(.venv) ~/dev/git/test $ sf code-analyzer run

Streaming logs in real time to:
    /var/folders/1q/7yjk0jjn6m33kf2h36kswnmc0000gn/T/sfca-2025_12_04_17_46_10_490.log

Selecting rules... done.
Executing rules... done. Executed rules from flow, regex.

Violation file paths relative to '/Users/rsussland/dev/git/test'.
  #   Severity       Rule                    Location                           Message
 ──────────────────────────────────────────────────────────
  1   3 (Moderate)   flow:SameRecordUpdate   test_trigger.flow-meta.xml:109:1   An AfterSave record trigger is modifying the same record. Record modifications should be done in BeforeSave triggers, not AfterSave triggers. The trigger definition may be in a parent flow that calls the current flow as a subflow, passing in the recordId of the trigger record.



=== Summary

Found 1 violation(s) across 1 file(s):
    1 Moderate severity violation(s) found.
